src/main/groovy/ui/GroovyMain.java

```diff
@@ -305,6 +310,7 @@ private static boolean process(CommandLine line) throws ParseException, IOExcept
         main.isScriptFile = !line.hasOption('e');
         main.debug = line.hasOption('d');
         main.conf.setDebug(main.debug);
+        main.conf.setParameters(line.hasOption("pa") || System.getProperty("parameters") != null);
```

Comment:
We wouldn't normally check system properties here but rather in CompilerConfiguration#configure. We'd also normally prefix properties e.g. like "groovy.parameters" unless Java is already using something without a java prefix in which case we might use directly what Java uses.